frappe/workflow/doctype/workflow/workflow.py

```diff
@@ -126,13 +126,15 @@ def set_active(self):
 
 @frappe.whitelist()
 def get_fieldnames_for(doctype):
+	frappe.has_permission(doctype=doctype, ptype='read', throw=True)
```

Comment:
This should utilize `frappe.model.get_permitted_fields` instead as it applies permlevel checks too.

Reply:
The use of field names from this method is for configuring workflow, I think we can leave it here. 

Really it's not even required cause meta is available without permission checks. :woozy_face:

Reply:
this function isn't even correct :woozy_face: comparing fieldname with fieldtypes. 

https://github.com/frappe/frappe/pull/20653